starfish/imagestack/imagestack.py

```diff
@@ -175,35 +159,39 @@ def __init__(
             dims=data_dimensions,
             coords=data_tick_marks,
         )
-        self._coordinates = xr.DataArray(
-            np.empty(
-                shape=coordinates_shape,
-                dtype=np.float32,
-            ),
-            dims=coordinates_dimensions,
-            coords=coordinates_tick_marks,
-        )
 
         all_selectors = list(self._iter_axes({Axes.ROUND, Axes.CH, Axes.ZPLANE}))
         first_selector = all_selectors[0]
         tile = tile_data.get_tile(r=first_selector[Axes.ROUND],
                                   ch=first_selector[Axes.CH],
                                   z=first_selector[Axes.ZPLANE])
-        # only compare X,Y coords
+
+        # Set up coordinates
+        self._data[Coordinates.X.value] = xr.DataArray(
+            np.linspace(tile.coordinates[Coordinates.X][0], tile.coordinates[Coordinates.X][1],
+                        self.xarray.sizes[Axes.X.value]), dims=Axes.X.value)
+        self._data[Coordinates.Y.value] = xr.DataArray(
+            np.linspace(tile.coordinates[Coordinates.Y][0], tile.coordinates[Coordinates.Y][1],
+                        self.xarray.sizes[Axes.Y.value]), dims=Axes.Y.value)
+        if Coordinates.Z in tile.coordinates:
+            # Fill with zeros for now, then replace with calculated midpoints
+            self._data[Coordinates.Z.value] = xr.DataArray(np.zeros(
+                self.xarray.sizes[Axes.ZPLANE.value]),
+                dims=Axes.ZPLANE.value)
+
+        # Get coords on first tile to verify imagestack is aligned
```

Comment:
This comment doesn't make sense to me. I imagine that _all_ tiles would need to be parsed to verify this.

Reply:
I use the coordinates from the first tile and verify that the rest match

Reply:
```suggestion
        # Get coords on first tile, then verify all subsequent tiles are aligned
```